File: java/mohit/Trees-BST/ValidateBST.java
static int minimum = Integer.MIN_VALUE;
static int solve(TreeNode root)
{
    //CODE HERE
    if(isBST(root)) return 1;
    else return 0;
}

static boolean isBST(TreeNode root)
{
    if(root == null) return true;

    if(!isBST(root.left)) return false;
    
    if(minimum < root.val) minimum = root.val;
    else return false;

    if(!isBST(root.right)) return false;

    return true;
}

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/ValidateBST.py
prev = None

def f(root):
    if root:
        if not f(root.left):
            return False
        global prev
        if prev >= root.val:
            return False
        prev = root.val
        if not f(root.right):
            return False
        return True
    return True


def solve(root):
    # CODE HERE
    global prev
    prev = -(2<<31+1)
    return 1 if f(root) else 0